src/beamlime/core/system.py

```diff
@@ -97,18 +158,11 @@ def kill(self) -> None:
             self._tasks.clear()
 
     def __del__(self) -> None:
-        while self.applications:
-            app_name, app_inst_gr = self.applications.popitem()
-            app_inst_gr.kill()
-            self.info("Deleting %s instance ... ", app_name)
-            del app_inst_gr
+        del self.applications
```

Comment:
Why is this explicit `del` of `applications` necessary?

Reply:
I forgot to delete this after using the debugger ...